util/job_manager.rb

```diff
@@ -96,6 +98,19 @@ def on_killed
       @wrapper&.kill
       @killed = true
     end
+
+    def populate_data(services_vrps)
+      points = services_vrps.flat_map{ |s_v| s_v[:vrp].points }.uniq
+      points.each{ |point| Models::Point.insert(point) }
+
+      rests = services_vrps.flat_map{ |s_v| s_v[:vrp].rests }.uniq
+      rests.each{ |rest| Models::Rest.insert(rest) }
+
+      services_vrps.each{ |s_v|
+        s_v[:vrp].vehicles.each{ |vehicle| Models::Vehicle.insert(vehicle) }
+        s_v[:vrp].services.each{ |service| Models::Service.insert(service) }
+      }
+    end
```

Comment:
It looks like we lost all the modifications you made for this function. If my memory serves me well there was  Relations and Units and the name was something like `load_models`. I don't remember if there was anything else, can you check?

Reply:
Oh there was the strategic delete_all. 

Also I remember this function being called in job_manager,rb (which is still here) and from wrapper_vrp.rb (maybe?) which doesn't exist anymore. Is that voluntary?

Reply:
Oh, As all the find_by_id were in one commit, i was assuming these changes were wrap together, it should be removed as well.

Reply:
I think the modification we did (which are not related to find_by_id) were legit and we can keep them.

Reply:
The delete_all/populate logic was called both synchronously and asynchronously if we have multiple repetitions.

Populate the database without using it, may only bring performance issues :/

Reply:
Then I fail to see why we need this populate logic now at all. All find_by_id logic is removed.. If we don't use the database, lets get rid of populate_data all together.

My thinking is that, if we want to introduce such a function we should do as a whole. If we populate only some objects and populate the database only on job_manager then a soft/hard-copy by id (or create by id relation) operation would work for the first repeat and fail on the second repeat if there is repetition. I don't think this is acceptable.

Reply:
I just saw that this (removing populate data all together) is what you did ;)